Source/Expressive/Functions/String/IndexOfFunction.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections;
+using System.Text;
+using Expressive.Expressions;
+
+namespace Expressive.Functions.String
+{
+    internal class IndexOfFunction : FunctionBase
+    {
+        #region FunctionBase Members
+
+        public override string Name
+        {
+            get
+            {
+                return "IndexOf";
+            }
+        }
+
+        public override object Evaluate(IExpression[] parameters, Context context)
+        {
+            this.ValidateParameterCount(parameters, -1, 2);
+
+            object value = parameters[0].Evaluate(Variables);
+
+            if (value is null)
+            {
+                return null;
+            }
+            else if (value is string)
+            {
+                //String comparison
+                //Can use an optional start index parameter
+
+                string text = value.ToString();
+
+                value = parameters[1].Evaluate(Variables);
+
+                if (value is null)
+                {
+                    return null;
+                }
+
+                string match = value.ToString();
+
+                if (parameters.Length > 2)
+                {
+                    int start = Convert.ToInt32(parameters[2].Evaluate(Variables));
+                    return text.IndexOf(match, start);

```

Comment:
I think we need to consider case-insensitive checking here. You can use the `context.EqualityStringComparison` property to use what has been configured by the consumer and apply it inside the `IndexOf` method.